### ISSUE TITLE
Unexpected token ILLEGAL while generating a parser object

### ISSUE THREAD

**Issue by zipavlin** [rzymek/meteor-pegjs#1]
Hello,

I am trying to generate a parser object, but there is an error returned:
client/myParser.pegjs:1:22: Unexpected token ILLEGAL

The same thing happens when generating a server-side parser object.

Besides that it seem it doesn't really matter what the content of .pegjs file is as this error is returned even for the simplest grammar.

I wonder if you might have an idea, what could be the problem?

Thanks for your help,
Žiga

**Comment by rzymek** [comment]
Could you post here your gram file?

**Comment by zipavlin** [comment]
Sure!

The original one was:

``````
// line block
start = line:(codeBlock / headerItem / taskItem / listItem / textItem) {
 return line;
}

// code block
codeBlock = t:tab* ("'''" / "```") {
 return {
  level: t.length,
  type: "code",
  typeCode: 4
 };
}

// task item
taskItem = t:tab* c:("[" bullet? space? "]") space* word+ {
 return {
  level: t.length,
  type: "task",
  typeCode: 3,
  content: c.map(function (item) {if (item) return item;}).join("")
 };
}

// list item
listItem = t:tab* c:(bullet / int+ "."? ")"?) space* word+ {
 return {
  level: t.length,
  type: "list",
  typeCode: 2,
  content: c.length > 1 ? c.join("") : c
 };
}

// header item
headerItem = t:tab* c:("#")+ space* word+ {
 return {
  level: t.length,
  type: "header",
  typeCode: 1,
  content: c.join("")
 }
}

// text item
textItem = t:tab* space* word+ {
 return {
  level: t.length,
  type: "text",
  typeCode: 0
 }
}

// BASIC STRUCTURE //
space = " "
lb = "\n"
tab = "\t" / "  "
bullet = [-0o*+]
char = [a-zA-Z()_#@*$.;,:!&{}\"]
int = [0-9]
word = w:(char / int)+ space? { return w.join(""); }
``````

I tried also without comments and with something like:

```
start = w:word+ {
    return w.join(" ")
}
```

**Comment by rzymek** [comment]
What meteor version are you using?

**Comment by zipavlin** [comment]
Meteor 1.1.0.3

**Comment by rzymek** [comment]
I cannot reproduce your problem. I've pasted your grammar in my test project (https://github.com/rzymek/meteor-pegjs/tree/master/workspace) and it works fine.

**Comment by zipavlin** [comment]
Firstly, thank you for your time and effort to check it out and to help me! 

And, wow, this is weird. I tried with your test project and it still doesn't work (the same error). I also tried to create new meteor project and it doesn't working either.

I am on  Win 8.1, running node 0.12.6 and npm 3.2.0 if it's relevant.

**Comment by rzymek** [comment]
I'm using Ubuntu. I'll try to run it on Windows.
Show me your full log with the error. Is there any stack trace?

**Comment by zipavlin** [comment]
OK, so I think I found what was the problem in my case.

I think the problem was in the name that is being added to compiled string (compiler.js, under line 8), that (in my case) started as "server\parserGrammar.pegjs = (function ...", so it seems it was using "full" path as a name of the function that is being "exported". This is also why there was always an error on line 1, because illegal characters in the name.

I tried "cleaning" the name string to only take filename, without extension as follows:

``` javascript
var name = compileStep.inputPath.replace(/^.*[/]([^/]+).pegjs$/, '$1');*
// two steps bellow were added to remove path and extension from filename
name = name.split("\\");
name = name[name.length - 1].replace(".pegjs", "");
```

**After those two additional steps everything works as it should! :)**
